freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsEventGenerationIdsUtil.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.GetEventGenerationIdsResponse;
+import com.google.common.annotations.VisibleForTesting;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UmsEventGenerationIds;
+
+@Component
+public class UmsEventGenerationIdsUtil {
+
+    @VisibleForTesting
+    enum EventMapping {
```

Comment:
changed to UmsEventGenerationIdsProvider. Extractor doesn't work for me because the class retrieves the values as well as converts it. hmm... maybe that suggests that I should split the class or have the caller invoke the GrpcUmsClient directly. I'll consider that as a future improvement.